### PR TITLE
Reformat installation instruction and add Gentoo

### PR DESCRIPTION
**Pull request by kant2002**
I extract each platform as separate section, so that block would be easier to read.
Also add instructions how to compile application on Gentoo.

### CODE REVIEW

**Comment by Northfear** [comment]
`sdl2-mixer` and `sdl2-ttf` should probably be the default ones for Gentoo tho

**Comment by kant2002** [comment]
Probably yes. Due to my first exposure, I first install plain `desktop/systemd` profile, and only then `desktop/plasma/systemd`, so this may explain my need to install additional dependencies.

File: README.md [hunk 1]
```diff
@@ -25,8 +25,30 @@ You are required to have at least a demo version of Heroes of Might and Magic 2
...
 
+## Debian-based
+Please go to `script/linux` directory and run **install_sdl_1.sh** or **install_sdl_2.sh** file.
+
+## Arch
+
+For Arch based Linux distributions use the command `sudo pacman -S sdl sdl_mixer`.
+
+## RedHat-based
+
```
Comment (ihhub) [inline README.md:39]: Please remove this empty line.

File: README.md [hunk 2]
```diff
@@ -25,8 +25,30 @@ You are required to have at least a demo version of Heroes of Might and Magic 2
...
 --------------------------
-Unix OSes need an explicit installation of SDL packages. Please go to `script/macos` or `script/linux` directory depending on your OS package and run **install_sdl_1.sh** or **install_sdl_2.sh** file. For Arch based Linux distributions use the command `sudo pacman -S sdl sdl_mixer` instead. RPM based distributions like Fedora/Red Hat use the commands `sudo yum/dnf install SDL*`. openSUSE supports the One-Click-Install via `SDL_mixer.ymp` file in `script/linux`. For MacOS we highly recommend to use SDL 2 as latest versions of MacOS do not fully support SDL 1.
+## Mac OS
+Please go to `script/macos` and run **install_sdl_1.sh** or **install_sdl_2.sh** file. For MacOS we highly recommend to use SDL 2 as latest versions of MacOS do not fully support SDL 1.
 
+## Debian-based
+Please go to `script/linux` directory and run **install_sdl_1.sh** or **install_sdl_2.sh** file.
+
+## Arch
+
```
Comment (ihhub) [inline README.md:35]: Please remove this empty line.

File: README.md [hunk 3]
```diff
@@ -25,8 +25,30 @@ You are required to have at least a demo version of Heroes of Might and Magic 2
...
+## RedHat-based
+
+RPM based distributions like Fedora/Red Hat use the commands `sudo yum install SDL*` or `sudo dnf install SDL*`.
+
+## OpenSUSE
+openSUSE supports the One-Click-Install via `SDL_mixer.ymp` file in `script/linux`. 
+
+## Gentoo
+
```
Comment (ihhub) [inline README.md:46]: Please remove this empty line.

File: README.md [hunk 4]
```diff
@@ -25,8 +25,30 @@ You are required to have at least a demo version of Heroes of Might and Magic 2
...
+If you have Gentoo, in addition to basic distribution you need to run following commands
```
Comment (ihhub) [inline README.md:47]: Since the title already states Gentoo I think we should rephrase it to `In addition to basic distribution you need to run following commands:`. What do you think?

File: README.md [hunk 5]
```diff
@@ -25,8 +25,30 @@ You are required to have at least a demo version of Heroes of Might and Magic 2
...
+```
+emerge --ask sdl-mixer
+emerge --ask sdl-ttf
```
Comment (ihhub) [inline README.md:50]: Is it possible to combine these 2 commands into one?

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @kant2002 , I left few comments here. Could you please take a look?

**Comment by kant2002** [comment]
@ihhub I deliberately do not include instructions how to install dependencies for SDL1. Let me know if you want these option to be highlighted.

**Comment by ihhub** [comment]
> @ihhub I deliberately do not include instructions how to install dependencies for SDL1. Let me know if you want these option to be highlighted.

I think it's fine not to mention SDL 1 as it's useful only for really old OSes.

**Comment by ihhub** [comment]
@kant2002 , thank you very much for this documentation cleanup!
